### PR TITLE
Add hook to allowing signing of Hexagon code

### PR DESCRIPTION
**Pull request by steven-johnson**
This allows a build tool to define an env var (HL_HEXAGON_CODE_SIGNER)
to sign Hexagon code; since Hexagon code is typically embedded as data,
it’s nontrivial to extract/sign/replace the code at a higher level in
the build system.

### CODE REVIEW

**Comment by abadams** [comment]
LGTM, but probably should wait for the windows branch builder to catch up.
I'm hoping the trouble pulling llvm 4.0 was a temporary glitch on the llvm
end.

**Comment by steven-johnson** [comment]
No rush to land; I'd like feedback from dsharletg@ and ronlieb@ before landing.

**Comment by ronlieb** [comment]
lgtm

File: src/HexagonOffload.cpp [hunk 1]
```diff
@@ -896,6 +896,43 @@ Buffer<uint8_t> compile_module_to_hexagon_shared_object(const Module &device_cod
...
+        //
+        // where unsigned and signed paths must not be the same file.
+        // If the signed file already exists, it will be overwritten.
+
+        TemporaryFile input("hvx_unsigned", ".so");
+        TemporaryFile output("hvx_signed", ".so");
+
+        debug(1) << "Signing Hexagon code: " << input.pathname() << " -> " << output.pathname() << "\n";
+
+        FILE *f = fopen(input.pathname().c_str(), "wb");
```
Comment (dsharletg) [inline src/HexagonOffload.cpp]: I think we should prefer std::fstream over C file IO.

**Pull request merged**